src/rules/correctReactImportRule.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2019-present, ProductBoard, Inc.
+ * All rights reserved.
+ */
+
+import * as Lint from "tslint";
+import * as ts from "typescript";
+
+type Configuration = {
+  reference: string;
+};
+
+export class Rule extends Lint.Rules.AbstractRule {
+  public static FAILURE_STRING =
+    "Don't import React with asterisk, use `import React from 'react';`";
+
+  public apply(sourceFile: ts.SourceFile): Array<Lint.RuleFailure> {
+    return this.applyWithWalker(
+      new CorrectReactImportWalker(sourceFile, this.getOptions())
+    );
+  }
+}
+
+class CorrectReactImportWalker extends Lint.RuleWalker {
+  private configuration: Configuration;
+
+  constructor(sourceFile: ts.SourceFile, options: Lint.IOptions) {
+    super(sourceFile, options);
+
+    const configuration = {
+      reference: "",
+      ...options.ruleArguments[0]
+    } as Configuration;
+
+    this.configuration = configuration;
+  }
+
+  private getFormattedError(error: string): string {
+    return (
+      error +
+      (this.configuration.reference ? ` ${this.configuration.reference}` : "")
+    );
+  }
+
+  public visitNamespaceImport(node: ts.NamespaceImport) {
+    if (node.name.text === "React") {
```

Comment:
Maybe we also need additional check, for the `from 'react'` part? 🤔

Reply:
Oh, GOD NO! XD

Reply:
Actually I changed this, if there is `*` import from `react` it's wrong.

e.g let's catch also `import * as ILoveReact from 'react'` 😈